models/reservation.js

```diff
@@ -106,45 +63,34 @@ exports.reserve = function(driverId, latitude, longitude) {
       reservation.attributes.longitude = spotLatLong.longitude;
       reservation.attributes.hostId = spot.attributes.userId;
       reservation.attributes.spotId = spot.id;
-      return reservation;
-    })
-    .then(createNewReservation)
-    .then(function(reservation) {
-      return Spot.addReservationId(reservation.attributes.spotId, reservation.id);
+      return reservationsRef.child(reservation.id).update(reservation.attributes);
```

Comment:
I think it's so weird that firebase's 'update' returns nothing instead of returning the updated reservation object...I wonder why that is